feat: add team ID validation to set-team-id command

intertwine opened the pull request:
# Add Team ID Validation to set-team-id Command

## Summary

Adds input validation to the `prime config set-team-id` command to ensure team IDs match the expected format before saving to configuration.

## Changes

### Implementation

- **Added validation function** (`validate_team_id()`) that checks team IDs match the pattern `^[a-zA-Z0-9]{25}$`
- **Updated `set_team_id` command** to validate input before persisting to config file
- **Clear error messaging** when invalid team ID format is provided
- **Preserves existing behavior** for empty strings (personal account mode)

### Testing

- **Created comprehensive test suite** (`test_config.py`) with 10 test cases covering:
  - Valid team IDs (lowercase, uppercase, mixed case)
  - Empty string handling (personal account)
  - Invalid formats (wrong length, special characters, spaces, etc.)
- **All tests passing** (12/12 including existing tests)

## Validation Rules

### ✅ Accepted Formats

- Exactly 25 alphanumeric characters (a-z, A-Z, 0-9)
- Example: `cmf0ohr9s0026testf3w68s5a`
- Empty string `""` (clears team ID for personal account)

### ❌ Rejected Formats

- Strings not exactly 25 characters long
- Strings containing special characters, spaces, or underscores
- Example: `intertest` (wrong length and format)

## Manual and Automated Testing

### Manual Testing

```bash
# Valid team ID - succeeds
uv run prime config set-team-id cmf0ohr9s0026testf3w68s5a
# Output: Team ID 'cmf0ohr9s0026testf3w68s5a' configured successfully!

# Invalid team ID - fails with clear error
uv run prime config set-team-id intertest
# Output: Error: Invalid team ID format. Team ID must be exactly 25 alphanumeric characters.

# Clear team ID - succeeds
uv run prime config set-team-id ""
# Output: Team ID cleared. Using personal account.
```

### Automated Tests

```bash
uv run pytest tests/test_config.py -v
# All 10 validation tests pass
```

## Files Changed

- `packages/prime/src/prime_cli/commands/config.py` - Added validation logic
- `packages/prime/tests/test_config.py` - New test file with comprehensive coverage

## Checklist

- [x] Code follows project style guidelines
- [x] Added comprehensive tests
- [x] All tests passing
- [x] Error messages are clear and helpful
- [x] Backward compatible (empty string still works)
- [x] No breaking changes to existing functionality


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds CUID v1-based validation for team IDs in the CLI and introduces tests covering valid/invalid cases.
> 
> - **CLI**:
>   - Add `TEAM_ID_PATTERN` and `validate_team_id()` in `packages/prime/src/prime_cli/commands/config.py` to enforce CUID v1 format (`^c[a-z0-9]{24}$`), allowing empty string.
>   - Integrate validation into `set_team_id` with clear error messaging and non-zero exit on invalid input.
> - **Tests**:
>   - New `packages/prime/tests/test_config.py` validating accepted empty string and rejecting uppercase, mixed case, wrong prefix, wrong length, spaces, special chars, and underscores.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 24d0a1dbd73d3fdb0943b0f7a238674e2c2f0982. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Jackmin801 commented on README.md:
````diff
@@ -114,6 +114,22 @@ Ensure you have at least two GPU to run the full test suite:
+
+
+```
+uv run accelerate launch -m lm_eval --model hf --model_args pretrained=CONVERTED_MODEL_PATH,add_bos_token=True,trust_remote_code=True  --tasks hellaswag --num_fewshot 10
````
The `trust_remote_code=True` isnt needed anymore

Jackmin801 reviewed (approved):
Nice! Just a small nit

intertwine committed:
feat: add team ID validation to set-team-id command

- Add regex validation for team IDs (25 alphanumeric characters)
- Reject invalid formats with clear error message
- Allow empty string for personal account
- Add comprehensive test suite with 10 test cases
- Validate examples: 'cmf0ohr9s0026ilerf3w68s6n' (valid), 'intertwine' (invalid)

AmeenP commented on packages/prime/src/prime_cli/commands/config.py:
```diff
@@ -9,6 +10,23 @@
 console = Console()
 
+# Team ID validation pattern: 25 alphanumeric characters
+TEAM_ID_PATTERN = re.compile(r"^[a-zA-Z0-9]{25}$")
```
this is a cuid() v1 so should be constrained to `r"^c[a-z0-9]{24}$"`

intertwine replied:
Duh, thanks - updated!

AmeenP commented:
Thanks for the contribution! Looks good after the minor fix

intertwine committed:
update input val to cuidv1
